common/metrics.py

```
@@ -0,0 +1,209 @@
+    
+    # Database query performance
+    database_query_duration = Histogram(
+        "database_query_duration_seconds", 
+        "Time spent querying database for cache data",
+        ["model", "status"],  # status: 'success', 'error'
+        buckets=[0.1, 0.5, 1.0, 5.0, 10.0, 15.0, 30.0, 60.0]
+    )
+
+    _instance: Union[DozensMetricsType, None] = None
+
+    def __new__(cls, *args, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for special method `__new__`

(ANN204)